specification/ai/Face/models.common.tsp

```diff
@@ -0,0 +1,196 @@
+  detection_01,
+
+  @doc("Detection model released in 2019 May with improved accuracy especially on small, side and blurry faces.")
+  detection_02,
+
+  @doc("Detection model released in 2021 February with improved accuracy especially on small faces.")
+  detection_03,
+}
+
+@doc("Model for object name and user data.")
+model NameAndUserData {
+  @doc("User defined name, maximum length is 128.")
+  name: string;
+
+  @doc("Optional user defined data. Length should not exceed 16KB.")
+  userData?: string;
+}
+
+@doc("Common model for persisted face.")
+@resource("persistedfaces")
+model BaseFace {
+  @key
+  @visibility("read")
+  @doc("Face ID of the face.")
+  persistedFaceId: string;
+
+  @doc("User-provided data attached to the face. The size limit is 1KB.")
+  userData?: string;
+}
+
+@doc("Query parameter for adding face to person.")
+model AddFaceOptions {
+  @query
+  @doc("A face rectangle to specify the target face to be added to a person, in the format of 'targetFace=left,top,width,height'.")
+  targetFace?: string;
+
+  @query
+  @doc("The 'detectionModel' associated with the detected faceIds. Supported 'detectionModel' values include 'detection_01', 'detection_02' and 'detection_03'. The default value is 'detection_01'.")
+  detectionModel?: DetectionModel;
+
+  @query
+  @doc("User-provided data attached to the face. The size limit is 1KB.")
+  userData?: string;
+}
+
+@doc("Model for add face by url.")
+model AddFaceFromUrlContent {
+  @doc("URL of input image.")
+  url: url;
+}
+
+@doc("Model for add face result.")
+model AddFaceResult {
+  @doc("The persisted face id.")
+  persistedFaceId: string;
+}
+
+@doc("Query parameters for whether response resource should include recognition model.")
+model ReturnRecognitionModelOptions {
+  @query
+  @doc("Return 'recognitionModel' or not. The default value is false.")
+  returnRecognitionModel?: boolean;
+}
+
+@doc("Training status of a container")
+model TrainingStatus {
+  @doc("Training status of the container.")
+  @lroStatus
+  status: TrainingStatusType;
+
+  @doc("A combined UTC date and time string that describes the created time of the person group, large person group or large face list.")
+  createdDateTime: utcDateTime;
+
+  @doc("A combined UTC date and time string that describes the last modify time of the person group, large person group or large face list, could be null value when the group is not successfully trained.")
+  lastActionDateTime: utcDateTime;
+
+  @doc("A combined UTC date and time string that describes the last successful training time of the person group, large person group or large face list.")
+  lastSuccessfulTrainingDateTime: utcDateTime;
+
+  @doc("Show failure message when training failed (omitted when training succeed).")
+  message?: string;
+}
+
+@doc("Type of training status.")
+enum TrainingStatusType {
```

Comment:
We already have a standard enum for LRO status. See https://azure.github.io/typespec-azure/docs/getstarted/azure-core/step06 for an example. The values you define here aren't correct, for example, and should be PascalCase.

Reply:
This is also for operation GAed a long time ago.